File: lib/src/communication/menu_registrar.dart
library geiger_api;

import 'package:geiger_localstorage/geiger_localstorage.dart';

import 'geiger_url.dart';

/// Registrar interface for Menus.
abstract class MenuRegistrar {
  /// Register a menu entry [menu] in the toolbox core.
  ///
  /// [menu] is an unstored, internationalized node reflecting a menu entry. The node path is an identifier which should be prefixed with :menu:<PluginUUID>:<menuIdentifier>.
  /// the following keys should be supported:
  /// - 'name' The name of the menu to appear (internationalizable)
  /// - 'tooltip' for the menu (internationalizable)
  /// - 'help' the help text (internationalizable)
  /// Must also provide the [action] to be triggered.
  Future<void> registerMenu(Node menu, GeigerUrl action);

  /// Enable a previously registered [menuId].
  Future<void> enableMenu(String menuId);

  /// Disable a previously registered [menuId].
  Future<void> disableMenu(String menuId);

  /// Removing a menu entry [menuId] from th toolbox core.
  Future<void> deregisterMenu(String menuId);
}

File: java/GeigerApiPluginConnector/src/main/java/eu/cybergeiger/communication/MenuRegistrar.java
package eu.cybergeiger.communication;

/**
 * <p>Registrar interface for Menus.</p>
 */
public interface MenuRegistrar {

  /**
   * <p>Register a Menu entry in the toolbox core.</p>
   *
   * <p>The method only allows URL with protocol GEIGER (geiger://).
   * The host reflects the plugin ID.</p>
   *
   * @param menu    menu string depicting menu hierarchy
   * @param action  action URL to be triggered
   */
  void registerMenu(String menu, GeigerUrl action);

  /**
   * <p>Enable a previously registered menu.</p>
   *
   * @param menu  menu string depicting menu hierarchy
   */
  void enableMenu(String menu);

  /**
   * <p>Disable a previously registered menu.</p>
   *
   * @param menu  menu string depicting menu hierarchy
   */
  void disableMenu(String menu);

  /**
   * <p>Removing a menu entry from th toolbox core.</p>
   *
   * <p>The method only allows URL with protocol GEIGER (geiger://).</p>
   *
   * @param menu    menu string depicting menu hierarchy
   */
  void deregisterMenu(String menu);

}
